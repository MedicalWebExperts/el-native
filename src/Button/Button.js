/* eslint no-console: ["error", { allow: ["error"] }] */
import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import Styles from './styles';

const propTypes = {
  style: PropTypes.object,
  /*
  disabled: PropTypes.bool,
  block: PropTypes.bool,
  size: PropTypes.oneOf("large", "medium", "small"),
  outline: PropTypes.bool,
  rounded: PropTypes.bool,
  transparent: PropTypes.bool,
  processing: PropTypes.bool,
  text: PropTypes.string,
  iconRight: PropTypes.string,
  iconLeft: PropTypes.string,
  icon: PropTypes.string,
  onPress: PropTypes.func.isRequired,
  */
};

const defaultProps = {
  style: {},
  disabled: PropTypes.bool,
  block: PropTypes.bool,
  size: PropTypes.oneOf('large', 'medium', 'small'),
  outline: PropTypes.bool,
  rounded: PropTypes.bool,
  transparent: PropTypes.bool,
  processing: PropTypes.bool,
  text: PropTypes.string,
  iconRight: PropTypes.string,
  iconLeft: PropTypes.string,
  icon: PropTypes.string,
  onPress: PropTypes.func.isRequired,
};

/**
 * Map component description.
 */
export default function Button(props) {
  return <View style={{ ...Styles.container, ...props.style.container }} />;
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
