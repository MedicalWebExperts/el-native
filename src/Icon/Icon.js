import React from 'react';
import { string, number } from 'prop-types';
import VectorIcon from 'react-native-vector-icons/Ionicons';

const propTypes = {
  /**
   * Color of the icon. Default: Theme white color.
   */
  color: string,
  /**
   * Size of the icon. Default: 16.
   */
  size: number,
  /**
   * What icon to show, see Icon Explorer app or one of the links above. Default: heart.
   */
  name: string,
};

const defaultProps = {
  color: 'white',
  size: 16,
  name: 'ios-heart',
};

const Icon = props => <VectorIcon {...props} />;

Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;

export default Icon;
