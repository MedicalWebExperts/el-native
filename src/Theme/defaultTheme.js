import buttonStyles from './components/buttonStyles';

export default {
  font: {
    size: 14,
    color: '#000',
  },
  ...buttonStyles,
};
