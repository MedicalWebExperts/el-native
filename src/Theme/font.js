import colors from './colors';

export default {
  size: 14,
  color: colors.text,
};
