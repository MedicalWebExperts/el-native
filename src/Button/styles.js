import Theme from '../Theme';

const theme = Theme.getTheme();

export default {
  container: {
    alignItems: 'flex-end',
  },
  iconWrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  textWithIconLeft: {
    marginLeft: 6,
  },
  textWithIconRight: {
    marginRight: 6,
  },
  button: {
    flexDirection: 'row',
    padding: 8,
    justifyContent: 'center',
    alignItems: 'center',
    ...theme.button,
  },
  text: {
    ...theme.buttonText,
  },
  buttonDisabled: {
    ...theme.buttonDisabled,
  },
  textDisabled: {
    ...theme.buttonDisabledText,
  },
};
