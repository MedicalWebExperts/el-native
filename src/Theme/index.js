import defaultTheme from './defaultTheme';

class Theme {
  constructor() {
    this.theme = defaultTheme;
  }
  setTheme(theme) {
    this.theme = { ...this.theme, ...theme };
    return this.theme;
  }
  getTheme() {
    return this.theme;
  }
}
const themeInstance = new Theme();

export default themeInstance;
