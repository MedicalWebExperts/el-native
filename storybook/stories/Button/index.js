import React from 'react';

import { storiesOf } from '@storybook/react-native';

import CenterView from '../CenterView';
import { Button } from '../../../src';

storiesOf('Button', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('Default', () => <Button />);
