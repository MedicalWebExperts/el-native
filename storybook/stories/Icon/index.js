import React from 'react';
import { storiesOf } from '@storybook/react-native';

import CenterView from '../CenterView';
import { Icon } from '../../../src';

storiesOf('Icon', module)
  .addDecorator(getStory => (
    <CenterView style={{ backgroundColor: '#AFAFAF' }}>{getStory()}</CenterView>
  ))
  .add('Default', () => <Icon />)
  .add('Custom color', () => <Icon size={200} color="aquamarine" />)
  .add('Custom size', () => <Icon size={150} />)
  .add('Custom icon', () => <Icon size={200} name="ios-basketball" />)
  .add('Custom all', () => <Icon name="ios-football" size={250} color="azure" />);
