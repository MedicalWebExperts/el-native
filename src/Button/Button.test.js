/* global describe it expect jest:true */
import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';
import Icon from '../Icon/Icon';

import {
  shouldHaveText,
  shouldHaveComponent,
  shouldSimulateOnPress,
  shouldHaveStyles,
} from '../utils/tests';
import Theme from '../Theme';

const theme = Theme.getTheme();
const styles = theme.button;

describe('Button', () => {
  it('should render a text', () => {
    shouldHaveText(<Button onPress={() => null} text="LOGIN" />, 'LOGIN');
  });

  it('should be disabled', () => {
    const wrapper = shallow(<Button onPress={() => null} text="LOGIN" disabled />);

    expect(wrapper.props().disabled).toBeTruthy();
  });

  it('should have disabled styles', () => {
    const defaultStyles = styles.default;
    const disabledStyles = styles.disabled;

    const wrapper = shallow(<Button onPress={() => null} text="LOGIN" disabled />);

    shouldHaveStyles(wrapper.find('View'), { ...defaultStyles, ...disabledStyles });
  });

  it('should have background styles', () => {
    const defaultStyles = styles.default;
    const bgStyles = { backgroundColor: 'red' };

    const wrapper = shallow(<Button onPress={() => null} text="LOGIN" backgroundStyles={bgStyles} />);

    shouldHaveStyles(wrapper.find('View'), { ...defaultStyles, ...bgStyles });
  });

  it('should be block', () => {
    const defaultStyles = styles.default;
    const blockStyles = { width: '100%' };

    const wrapper = shallow(<Button onPress={() => null} text="LOGIN" block />);

    shouldHaveStyles(wrapper.find('View'), { ...defaultStyles, ...blockStyles });
  });

  it('should be transparent', () => {
    const defaultStyles = styles.default;
    const transparentStyles = { backgroundColor: 'transparent', elevation: 0 };

    const wrapper = shallow(<Button onPress={() => null} text="LOGIN" transparent />);

    shouldHaveStyles(wrapper.find('View'), { ...defaultStyles, ...transparentStyles });
  });

  it('should be outlined', () => {
    const defaultStyles = styles.default;
    const customStyles = { backgroundColor: 'transparent', elevation: 1 };

    const wrapper = shallow(<Button onPress={() => null} text="LOGIN" outline />);

    shouldHaveStyles(wrapper.find('View'), { ...defaultStyles, ...customStyles });
  });

  it('should be rounded', () => {
    const roundedDimensions = 40;
    const defaultStyles = styles.default;
    const customStyles = {
      width: roundedDimensions,
      height: roundedDimensions,
      borderRadius: parseInt(roundedDimensions / 2, 10),
    };

    const wrapper = shallow(<Button onPress={() => null} text="LOGIN" roundedDimensions={40} />);

    shouldHaveStyles(wrapper.find('View'), { ...defaultStyles, ...customStyles });
  });

  it('should render an Icon', () => {
    shouldHaveComponent(<Button onPress={() => null} icon="ios-basketball" />, Icon);
  });

  it('should render an Left Icon', () => {
    shouldHaveComponent(<Button onPress={() => null} iconLeft="ios-basketball" />, Icon);
  });

  it('should render an Right Icon', () => {
    shouldHaveComponent(<Button onPress={() => null} iconRight="ios-basketball" />, Icon);
  });

  it('should render a Spinner', () => {
    shouldHaveComponent(<Button onPress={() => null} processing />, 'ActivityIndicator');
  });

  test('should call the onPress', () => {
    shouldSimulateOnPress(<Button onPress={() => null} />);
  });
});
