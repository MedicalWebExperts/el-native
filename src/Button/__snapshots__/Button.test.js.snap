// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshots should render a Button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel=""
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#007AFF",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Button Snapshots should render a Button block 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel=""
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 8,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#007AFF",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      BLOCK
    </Text>
  </View>
</View>
`;

exports[`Button Snapshots should render a Button disabled 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel=""
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#007AFF",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      DISABLED
    </Text>
  </View>
</View>
`;

exports[`Button Snapshots should render a Button with a left icon 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel=""
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "elevation": 0,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "padding": 8,
        "width": 40,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 18,
              "textAlign": "center",
            },
            Object {
              "marginLeft": 6,
            },
          ]
        }
      >
        LOGIN
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Snapshots should render a Button with a right button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel=""
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "elevation": 0,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "padding": 8,
        "width": 40,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 18,
              "textAlign": "center",
            },
            Object {
              "marginRight": 6,
            },
          ]
        }
      >
        LOGIN
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Snapshots should render a Button with a spinner 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel=""
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "elevation": 0,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "padding": 8,
        "width": 40,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="black"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
</View>
`;

exports[`Button Snapshots should render a Button with a spinner and text 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel=""
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "elevation": 0,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "padding": 8,
        "width": 40,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="black"
        hidesWhenStopped={true}
        size="small"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 18,
              "textAlign": "center",
            },
            Object {
              "marginLeft": 6,
            },
          ]
        }
      >
        LOADING...
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Snapshots should render a Button with custom styles 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel=""
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FF0000",
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#007AFF",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      STYLED
    </Text>
  </View>
</View>
`;

exports[`Button Snapshots should render a outline Button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel=""
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "elevation": 1,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "padding": 8,
        "width": 40,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      OUTLINE
    </Text>
  </View>
</View>
`;

exports[`Button Snapshots should render a rounded Button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel=""
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "elevation": 0,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "padding": 8,
        "width": 40,
      }
    }
  >
    <View
      style={
        Object {
          "padding": undefined,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Snapshots should render a transparent Button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel=""
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "elevation": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 8,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      TRANSPARENT
    </Text>
  </View>
</View>
`;

exports[`Button Snapshots should render an icon Button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel=""
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "elevation": 0,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "padding": 8,
        "width": 40,
      }
    }
  >
    <View
      style={
        Object {
          "padding": undefined,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
